Packs/ThousandEyes/Integrations/ThousandEyes/ThousandEyes.py

```diff
@@ -0,0 +1,807 @@
+        data: Optional[Dict] = None,
+        json_data: Optional[Dict] = None,
+        timeout: float = 10,
+        resp_type: str = "response",
+        test_module: bool = False
+    ) -> Union[Response, Dict]:
+        """
+            Handles the reached rate limit, performs API request to the specified endpoint and reutrns the full Response object
+
+        Args:
+            method (str) required: The HTTP method, for example, GET, POST, and so on.
+            url_suffix (str) required: The API endpoint.
+            params (dict): URL parameters to specify the query. Default is None.
+            data (dict): The data to send in a 'POST' request. Default is None.
+            json_data (dict): The dictionary to send in a 'POST' request. Default is None.
+            timeout: (float): Time (in seconds) for the client to wait to establish a connection before a timeout occurs.
+                                Default is 10.
+            resp_type (str): Determines which data format to return from the HTTP request. Other options are 'text',
+                             'content', 'xml' or 'response'. Use 'response' to return the full response object.
+                             Default is response.
+            test_module (bool): Boolean flag to check if function is called from test-module or not
+
+        Returns:
+            Either a Response Object or Dictionary, depending on the resp_type value
+        """
+        # return self._http_request(
+        #     method=method,
+        #     url_suffix=url_suffix,
+        #     params=params,
+        #     data=data,
+        #     json_data=json_data,
+        #     timeout=timeout,
+        #     resp_type=resp_type,
+        # )
+
+        if test_module:
+            response = self._http_request(
+                method=method,
+                url_suffix=url_suffix,
+                resp_type=resp_type
+            )
+
+            return response
+
+        try:
+            while True:
+                response = self._http_request(
+                    method=method,
+                    url_suffix=url_suffix,
+                    params=params,
+                    data=data,
+                    json_data=json_data,
+                    timeout=timeout,
+                    resp_type=resp_type,
+                )
+
+                response_headers = response.headers
+                # org_rate_limit = response_headers.get('x-organization-rate-limit-limit')
+                org_rate_limit_remaining = response_headers.get('x-organization-rate-limit-remaining')
+                org_rate_limit_reset = response_headers.get('x-organization-rate-limit-reset')
+
+                if response.status_code == 200 and int(org_rate_limit_remaining) > 0:
+                    return response
+
+                elif org_rate_limit_remaining == 0:
+                    current_timestamp = int(time.time())
+                    time.sleep(org_rate_limit_reset - current_timestamp)
```

Comment:
We require this step to handle the enforced rate limit by the API, so that in case a user hits it at any time, the script simply waits for the specified duration and retries the request.

It is not specific to one command, all the integration commands and subsequent API requests need to be able to handle that. Could you please guide as to how we can achieve the same using "schedule method"?

Reply:
Lets talk about this in the demo.